example/src/screens/GraphPage.tsx

```diff
@@ -19,15 +20,49 @@ export function GraphPage() {
   const [isAnimated, setIsAnimated] = useState(true)
   const [enablePanGesture, setEnablePanGesture] = useState(true)
   const [enableFadeInEffect, setEnableFadeInEffect] = useState(false)
+  const [enableCustomSelectionDot, setEnableCustomSelectionDot] =
+    useState(false)
+  const [enableRange, setEnableRange] = useState(false)
+  const [enableIndicator, setEnableIndicator] = useState(false)
+  const [indicatorPulsating, setIndicatorPulsating] = useState(false)
 
   const [points, setPoints] = useState(() => generateRandomGraphData(POINTS))
-  const smallPoints = generateSinusGraphData(9)
+  const smallPoints = useMemo(() => generateSinusGraphData(9), [])
 
   const refreshData = useCallback(() => {
     setPoints(generateRandomGraphData(POINTS))
     hapticFeedback('impactLight')
   }, [])
 
+  const highestDate = useMemo(
+    () =>
+      points.length !== 0 && points[points.length - 1] != null
+        ? points[points.length - 1]!.date
+        : undefined,
+    [points]
+  )
+  const range: GraphRange | undefined = useMemo(() => {
+    let r: GraphRange | undefined
+
+    if (enableRange) {
+      r = {
+        y: {
+          min: -200,
+          max: 200,
+        },
+      }
+
+      if (points.length !== 0 && highestDate != null) {
+        r.x = {
+          min: points[0]!.date,
+          max: new Date(highestDate.getTime() + 30),
+        }
+      }
+    }
+
+    return r
+  }, [enableRange, highestDate, points])
```

Comment:
I think this looks cleaner:
```suggestion
  const range: GraphRange | undefined = useMemo(() => {
    // if range is disabled, default to infinite range (undefined)
    if (!enableRange) return undefined
    
    if (points.length !== 0 && highestDate != null) {
      return {
        x: {
          min: points[0]!.date,
          max: new Date(highestDate.getTime() + 30),
        },
        y: {
          min: -200,
          max: 200,
        }
      }
    } else {
      return {
        y: {
          min: -200,
          max: 200
        }
      }
    }
  }, [enableRange, highestDate, points])
```

What do you think? (Hoping the logic is still the same, I haven't tested this!)